pos_container_deposit/models/product_template.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 Hunki Enterprises BV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    is_deposit = fields.Boolean(
+        help="Check this if this product is a container for which you charge deposit",
+    )
+    deposit_product_id = fields.Many2one(
+        "product.product",
+        "Deposit product",
+        domain=[("is_deposit", "!=", False)],
+        help="If this product is packaged in a container for which you charge deposit, "
+        "add a product here that stands for the deposit",
+        compute="_compute_deposit_product_id",
+        inverse="_inverse_deposit_product_id",
+        search="_search_deposit_product_id",
+    )
+
+    @api.depends("product_variant_ids.deposit_product_id")
+    def _compute_deposit_product_id(self):
+        for this in self:
+            this.deposit_product_id = this.product_variant_ids.deposit_product_id
+
+    def _inverse_deposit_product_id(self):
+        for this in self:
+            this.product_variant_ids.write(
+                {
+                    "deposit_product_id": this.deposit_product_id,
+                }
+            )
+
+    def _search_deposit_product_id(self, operator, value):
+        return [("product_variant_ids.deposit_product_id", operator, value)]
+
+    def copy(self, default=None):
+        """
+        Take care that copies include the deposit product
+        """
+        if default is None:
+            default = {}
+        default.setdefault("deposit_product_id", self.deposit_product_id.id)
+        return super().copy(default=default)
```

Comment:
I'm surprised this is needed 🤔 

doesn't the ORM just copies the variant fields, and this field with it?
additionally, this isn't a stored field, so double surprise

Reply:
it is not stored, that's where the problem comes from. products copy [actually the template](https://github.com/OCA/OCB/blob/16.0/addons/product/models/product_product.py#L395), and then return its variant, so we need to trigger the inverse function on the product template manually because non-stored fields aren't copied usually

Reply:
Thanks for clarifying!